src/xmpp/stanza.c

```diff
@@ -1201,11 +1210,17 @@ stanza_get_delay(xmpp_stanza_t *const stanza)
                 GDateTime *utc_datetime = g_date_time_new_from_timeval_utc(&utc_stamp);
                 GDateTime *local_datetime = g_date_time_to_local(utc_datetime);
                 g_date_time_unref(utc_datetime);
+                if (from) {
+                    *from = g_strdup(xmpp_stanza_get_attribute(x, STANZA_ATTR_FROM));
+                }
                 return local_datetime;
             }
         }
     }
 
+    if (from) {
```

Comment:
Why do we need this?

Reply:
As I mentioned in the room, this patch is just a quick hack to test/show the required behavior change, it's not necessarily meant to be merged as-is.  The patch needs two values returned from `stanza_get_delay()` (the timestamp and the `from`), and from a quick glance at the code it wasn't obvious to me how you typically do that.  So I just went for returning `from` via an out parameter, which the caller can set to `NULL` if he's not interested in it.  `if (from)` checks whether it's `NULL`.

Reply:
Yes, but why would we need to set it to `NULL`  if it's set?

Reply:
Ok I see it now. We could also do `gchar *from = NULL` so the `g_free(from);` is okay, but your way is saver.

Reply:
Ah.  This is the case where we didn't find a `from` tag because we didn't find a delay attribute at all.  Could be omitted if we're sure no caller will ever look at `from` if the function returned `NULL` (i.e. right now it could safely be omitted).